cpp/src/arrow/array.cc

```diff
@@ -447,6 +447,24 @@ std::shared_ptr<DataType> FixedSizeListArray::value_type() const {
 
 std::shared_ptr<Array> FixedSizeListArray::values() const { return values_; }
 
+Status FixedSizeListArray::FromArrays(const std::shared_ptr<Array>& values,
+                                      int32_t list_size, std::shared_ptr<Array>* out) {
+  if (list_size < 0) {
+    return Status::Invalid("list_size needs to be a positive integer");
+  }
+  if ((values->length() % list_size) != 0) {
```

Comment:
This is obviously problematic when `list_size` is 0. You probably want to add a dedicated code path for that.

Reply:
Hmm, that's right. 
But, I don't really see how to construct an array with list_size of 0 in this function, as you would need to know the length of the final array. But this cannot be derived from the length of the values array, since this will have no values for a list_size of 0. So maybe raising an error is fine?

Reply:
Yes, raising an error seems reasonable.